python/ray/train/_internal/utils.py

```diff
@@ -58,6 +59,23 @@ def check_for_failure(
                 return False, exc
             except Exception as exc:
                 # Other (e.g. training) errors should be directly raised
+                # If exception is raised in the serialization module,
+                # we guide the user to look for that
+                # test_data_parallel_trainer.py::test_serialization_errors
+                # tests that this is being printed
+                if "serialization.py" in traceback.format_exc():
+                    logger.error(
+                        "An exception raised here from the serialization module "
+                        "is most likely caused by an issue with deserialization "
+                        "(eg. with Torch models or tensors). "
+                        "Ensure that you are reporting a Checkpoint type specific "
+                        "to the framework you are using (eg. `TorchCheckpoint` if "
+                        "checkpointing a Torch model). Those Checkpoint types "
+                        "contain special serialization/deserialization logic "
+                        "that helps avoid deserialization exceptions. "
+                        "No special handling logic is applied for objects "
+                        "passed in the `metrics` dict in the `report()` method!"
+                    )
```

Comment:
I'm unclear on the purpose of this error message.

Is it to warn users against using the base `Checkpoint` class? If so, then aren't we already doing that in `_warn_about_bad_checkpoint_type`?

Or if it's for more general purpose serialization, then I think this message could be misleading. Like even in `test_serialization_errors` test, we would still run into serialization issues even if we use framework specific checkpoints right?

Reply:
This warning is intended to guide the user towards the most likely cause of an otherwise uninformative exception that can appear due to issues with serialization we do not account for in framework-specific checkpoints or if the bad object is passed as a metric. If you think it just makes things more confusing, I am happy to remove it!

Reply:
Removed for now